framework/src/headers/ip/v6/icmp/neighbor_options.rs

```diff
@@ -0,0 +1,206 @@
+use super::{EndOffset, Ipv6VarHeader};
+use headers::mac::MacAddress;
+use num::FromPrimitive;
+use std::default::Default;
+use std::fmt;
+use std::marker::PhantomData;
+
+#[derive(FromPrimitive, Debug, PartialEq)]
+#[repr(u8)]
+pub enum Icmpv6OptionType {
+    SourceLinkLayerAddress = 1,
+    TargetLinkLayerAddress = 2,
+    PrefixInformation = 3,
+    RedirectHeader = 4,
+    MTU = 5,
+}
+
+impl fmt::Display for Icmpv6OptionType {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            Icmpv6OptionType::SourceLinkLayerAddress => write!(f, "Source Link Layer Address"),
+            Icmpv6OptionType::TargetLinkLayerAddress => write!(f, "Target Link Layer Address"),
+            Icmpv6OptionType::PrefixInformation => write!(f, "Prefix Information"),
+            Icmpv6OptionType::RedirectHeader => write!(f, "Redirect Header"),
+            Icmpv6OptionType::MTU => write!(f, "MTU"),
```

Comment:
there are other option types (gotta follow the trail of RFCs).  Things we saw in wireshark like "Default DNS Resolver" iirc.  What would the result be if we got an option type we do not know?

Reply:
Default DNS Resolver isn't part of the spec. You can configure extra options as you see fit. Your router would have to know to look for them

Reply:
I guess my question was, what happens if the system sees a "DNS Resolver" or some other option type that we do not presently support?

Should we have an "Unsupported" option type that we fall back to?

This specific piece of code is for `Display`, I am thinking more generally

Reply:
@pauljamescleary i think if we see an option type we dont support, we dont try to parse it. we'll move onto the next option in the list